### ISSUE TITLE
Segmentation Fault when try re.eval("library('forecast')")

### ISSUE THREAD

**Issue by rizer1980** [s-u/rJava#313]
Hello.

When I try to include some packages from java in R, the process crashes with Segmentation Fault.
Reproduce:
add one of this lines to (https://github.com/s-u/rJava/tree/master/jri/examples/rtest.java)

> re.eval("library('forecast')");
> re.eval("library('tsDyn')");

Get:

> Rengine.eval(library('forecast')): BEGIN Thread[main,5,main]
> Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

Ubuntu WSL2. R version 4.3.1 stable. Tried different java versions 11 and 17. Tried to compile R from sources with --enable-R-shlib.
The packages are installed in R, and they connect normally from R itself.
I guess that the problem is somehow connected with libraries, but I can’t imagine how to debug such a problem.

Thx.

**Comment by s-u** [comment]
@rizer1980 I cannot reproduce the problem:

```
$ git diff
diff --git a/jri/REngine b/jri/REngine
--- a/jri/REngine
+++ b/jri/REngine
@@ -1 +1 @@
-Subproject commit 68b0147275919401084a9e1ebf6b576547a49931
+Subproject commit 68b0147275919401084a9e1ebf6b576547a49931-dirty
diff --git a/jri/examples/rtest.java b/jri/examples/rtest.java
index 85f9480..c550625 100644
--- a/jri/examples/rtest.java
+++ b/jri/examples/rtest.java
@@ -79,6 +79,9 @@ public class rtest {
                        to accomplish what you want */
                try {
                        REXP x;
+                       re.eval("library('forecast')");
+                       re.eval("library('tsDyn')");
+
                        re.eval("data(iris)",false);
                        System.out.println(x=re.eval("iris"));
                        // generic vectors are RVector to accomodate names
$ (cd examples && ${JAVA_HOME}/bin/javac -cp ../JRI.jar:. rtest.java)
Note: rtest.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
$ sh run rtest
Creating Rengine (with arguments)
[...]
Rengine created, waiting for R
Registered S3 method overwritten by 'quantmod':
  method            from
  as.zoo.data.frame zoo 
[...]
```
 Can you provide more details about your platform, versions (system, R, packages), etc.? Add `re.eval("print(sessionInfo())");` and compare it to your R session - common problem is using wrong package library in JRI due to the lack of correct env vars which is incompatible with the R you run.

**Comment by rizer1980** [comment]
Thanks for reply.

I created a pure wsl distribution from https://cloud-images.ubuntu.com/wsl/jammy/current/ and
tried to compile R-stable(2023-06-16), R-patched and R-devel from sources.
As a result, it worked without errors only in R-devel.
I couldn’t figure out why even R-devel crashes with seg.fault in my working wsl.
I tried all the BLAS/LAPACK options - it didn't help.
Also, I checked R-devel on a fresh Manjaro (KDE) - it works.

I also ran `strace sh rtest run` - it crashes with an error when I try to read pillar.rdx from the pillar library.
The funny thing is that if you load pillar first, like this:

>              re.eval("library('pillar')");
>              re.eval("library('tsDyn')");

then everything works fine)
I've settled on this option for now.
